ngraph/frontend/onnx_import/src/op/upsample.cpp

```diff
@@ -29,180 +29,144 @@ namespace ngraph
+                                                      const Output<ngraph::Node>& scales,
+                                                      const std::string& mode)
                 {
-                    InterpolateMode result = InterpolateMode::nearest;
-                    if (mode_str == "linear")
-                    {
-                        result = InterpolateMode::linear_onnx;
-                    }
-                    return result;
+                    const auto shape_of_data = std::make_shared<default_opset::Convert>(
+                        std::make_shared<default_opset::ShapeOf>(data), ngraph::element::f32);
+                    const auto multiply =
+                        std::make_shared<default_opset::Multiply>(shape_of_data, scales);
+                    const auto output_shape = std::make_shared<default_opset::Convert>(
+                        std::make_shared<default_opset::Floor>(multiply), ngraph::element::i64);
+
+                    return {std::make_shared<default_opset::Interpolate>(
+                        data, output_shape, scales, get_attributes(mode))};
                 }
-            }
+
+            } // namespace
 
             namespace set_1
             {
                 OutputVector upsample(const onnx_import::Node& node)
                 {
-                    const auto inputs = node.get_ng_inputs();
-                    const auto data = inputs.at(0);
-
-                    const auto data_shape = data.get_partial_shape();
-
-                    const auto scales = node.get_attribute_value<std::vector<float>>("scales");
+                    const auto height_scale = node.get_attribute_value<float>("height_scale");
+                    const auto width_scale = node.get_attribute_value<float>("width_scale");
                     const auto mode = node.get_attribute_value<std::string>("mode", "nearest");
-                    check_mode_support(node, mode);
+                    check_mode_support(node, mode, version_1);
 
-                    auto attrs = ngraph::op::v4::Interpolate::InterpolateAttrs();
-                    attrs.mode = convert_mode(mode);
-                    attrs.shape_calculation_mode = ShapeCalcMode::scales;
-                    attrs.coordinate_transformation_mode = Transform_mode::half_pixel;
-                    attrs.nearest_mode = Nearest_mode::round_prefer_floor;
-                    attrs.antialias = false;
-                    attrs.cube_coeff = -0.75;
+                    const auto data = node.get_ng_inputs().at(0);
 
-                    if (attrs.mode == InterpolateMode::linear_onnx)
-                    {
-                        attrs.coordinate_transformation_mode = Transform_mode::asymmetric;
-                    }
+                    static const std::string expectation{"Input tensor is required to be 4D."};
+                    const auto rank = data.get_partial_shape().rank();
+                    CHECK_VALID_NODE(node, rank.is_static(), expectation);
```

Comment:
A single call to `CHECK_VALID_NODE` with both checks in it would also be fine.

Reply:
My aim was to throw the same exception message in both cases. If only rank's length was tested and rank was dynamic, then some generic message would be thrown.